java.lang.IllegalArgumentException in JobIntentService on Android 8.1

I updated my app to use SDK 3.10.2. I'm seeing some issues on Android 8.1 and 8.0 on different devices  with the following exception
```java
java.lang.RuntimeException: 
  at android.app.ActivityThread.handleReceiver (ActivityThread.java:3208)
  at android.app.ActivityThread.-wrap17 (Unknown Source)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1684)
  at android.os.Handler.dispatchMessage (Handler.java:106)
  at android.os.Looper.loop (Looper.java:172)
  at android.app.ActivityThread.main (ActivityThread.java:6590)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:438)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:807)
Caused by: java.lang.IllegalStateException: 
  at android.app.ContextImpl.startServiceCommon (ContextImpl.java:1521)
  at android.app.ContextImpl.startService (ContextImpl.java:1477)
  at android.content.ContextWrapper.startService (ContextWrapper.java:650)
  at com.onesignal.JobIntentService$CompatWorkEnqueuer.enqueueWork (JobIntentService.java:204)
  at com.onesignal.JobIntentService.enqueueWork (JobIntentService.java:534)
  at com.onesignal.q.a (NotificationBundleProcessor.java:465)
  at com.onesignal.q.a (NotificationBundleProcessor.java:430)
  at com.onesignal.GcmBroadcastReceiver.a (GcmBroadcastReceiver.java:125)
  at com.onesignal.GcmBroadcastReceiver.onReceive (GcmBroadcastReceiver.java:72)
  at android.app.ActivityThread.handleReceiver (ActivityThread.java:3201)
```
Already included these line in manifest 
```xml
<service
   android:name=".util.NotificationExtender"
   android:permission="android.permission.BIND_JOB_SERVICE"
   android:exported="false">
   <intent-filter>
       <action android:name="com.onesignal.NotificationExtender" />
   </intent-filter>
</service>
```
@jkasten2

--- a/OneSignalSDK/onesignal/src/main/java/com/onesignal/JobIntentService.java
+++ b/OneSignalSDK/onesignal/src/main/java/com/onesignal/JobIntentService.java
@@ -23,6 +23,8 @@ package com.onesignal;
 // This allows using ether a IntentService even for Android 8.0 (API 26) when needed.
 //    - This allows starting NotificationExtenderService on a high priority FCM message
 //    - Even when the device is in doze mode.
+// Has fallback handling to job service if IllegalStateException is thrown when starting service.
+// ==== Testing ====
 // - The following can be used to confirm;
 //   - adb shell dumpsys deviceidle force-idle
 // - And the following to undo
@@ -314,7 +316,7 @@ abstract class JobIntentService extends Service {
 
         @Override
         public boolean onStopJob(JobParameters params) {
-            if (DEBUG) Log.d(TAG, "onStartJob: " + params);
+            if (DEBUG) Log.d(TAG, "onStopJob: " + params);
             boolean result = mService.doStopCurrentWork();
             synchronized (mLock) {
                 // Once we return, the job is stopped, so its JobParameters are no
@@ -531,7 +533,21 @@ abstract class JobIntentService extends Service {
         synchronized (sLock) {
             WorkEnqueuer we = getWorkEnqueuer(context, component, true, jobId, useWakefulService);
             we.ensureJobId(jobId);
-            we.enqueueWork(work);
+
+            // Can throw on API 26+ if useWakefulService=true and app is NOT whitelisted.
+            // One example is when an FCM high priority message is received the system will
+            // temporarily whitelist the app. However it is possible that it does not end up getting
+            //    whitelisted so we need to catch this and fall back to a job service.
+            try {
+                we.enqueueWork(work);
+            } catch (IllegalStateException e) {
+                if (useWakefulService) {
+                    we = getWorkEnqueuer(context, component, true, jobId, false);
+                    we.enqueueWork(work);
+                }
+                else
+                    throw e;
+            }
         }
     }
 

